src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2510,20 +2564,69 @@ public void Inheritance_07()
     public abstract int X { get; }
     public virtual int Y { get; }
 }
-data class B(int X, int Y) : A
+abstract data class B1(int X, int Y) : A
+{
+}
+data class B2(int X, int Y) : A
 {
 }";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics(
-                // (6,12): error CS0534: 'B' does not implement inherited abstract member 'A.X.get'
-                // data class B(int X, int Y) : A
-                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "B").WithArguments("B", "A.X.get").WithLocation(6, 12));
+                // (9,12): error CS0534: 'B2' does not implement inherited abstract member 'A.X.get'
+                // data class B2(int X, int Y) : A
+                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "B2").WithArguments("B2", "A.X.get").WithLocation(9, 12));
+
+            AssertEx.Equal(new string[0], GetProperties(comp, "B1").ToTestDisplayStrings());
+            AssertEx.Equal(new string[0], GetProperties(comp, "B2").ToTestDisplayStrings());
         }
 
         [Fact]
         public void Inheritance_08()
         {
             var source =
+@"abstract class A
+{
+    public abstract int X { get; }
+    public virtual int Y { get; }
+    public virtual int Z { get; }
+}
+abstract class B : A
+{
+    public override abstract int Y { get; }
+}
+data class C(int X, int Y, int Z) : B
+{
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (11,12): error CS0534: 'C' does not implement inherited abstract member 'A.X.get'
+                // data class C(int X, int Y, int Z) : B
+                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "C").WithArguments("C", "A.X.get").WithLocation(11, 12),
+                // (11,12): error CS0534: 'C' does not implement inherited abstract member 'B.Y.get'
+                // data class C(int X, int Y, int Z) : B
+                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "C").WithArguments("C", "B.Y.get").WithLocation(11, 12));
+
+            var actualMembers = GetProperties(comp, "C").ToTestDisplayStrings();
+            AssertEx.Equal(new string[0], actualMembers);
+        }
+
+        [Fact]
+        public void Inheritance_09()
+        {
+            var source =
+@"abstract data class C(int X, int Y)
+{
+    public abstract int X { get; }
+    public virtual int Y { get; }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
```

Comment:
>VerifyDiagnostics [](start = 17, length = 17)

This should be an error because `C` contains both a declared property `X` and a synthesized one, because, from [https://github.com/dotnet/csharplang/pull/3527](https://github.com/dotnet/csharplang/pull/3527)

'Members are synthesized unless an accessible concrete (non-abstract) member with a "matching" signature is either inherited or declared in the record body.'

Reply:
I assume you will work with @agocke to resolve the conflict (I expect by modifying the spec).

---
In reply to: [434224682](https://github.com/dotnet/roslyn/pull/44705#discussion_r434224682) [](ancestors = 434224682)